File: 2237_count_positions_on_street_with_required_brightness/solution.go
// LeetCode 2237 - Count Positions on Street With Required Brightness
// https://leetcode.com/problems/count-positions-on-street-with-required-brightness/

func meetRequirement(n int, lights [][]int, requirement []int) int {
	diff := make([]int, n+1)
	for _, light := range lights {
		pos, r := light[0], light[1]
		l := pos - r
		if l < 0 {
			l = 0
		}
		rr := pos + r
		if rr >= n {
			rr = n - 1
		}
		diff[l]++
		diff[rr+1]--
	}
	ans, cur := 0, 0
	for i := 0; i < n; i++ {
		cur += diff[i]
		if cur >= requirement[i] {
			ans++
		}
	}
	return ans
}

File: 2237_count_positions_on_street_with_required_brightness/Solution.kt
// LeetCode 2237 - Count Positions on Street With Required Brightness
// https://leetcode.com/problems/count-positions-on-street-with-required-brightness/

class Solution {

    fun meetRequirement(n: Int, lights: Array<IntArray>, requirement: IntArray): Int {

            var diff = IntArray(n + 1)
            for (light in lights) {
                var pos = light[0]; var r = light[1]
                var l = maxOf(0, pos - r)
                var rr = minOf(n - 1, pos + r)
                diff[l]++
                diff[rr + 1]--
            }
            var ans = 0; var cur = 0
            for (i in 0 until n) {
                cur += diff[i]
                if (cur >= requirement[i]) ans++
            }
            return ans

    }

}
